From d113096c77ce44b639ac888c0024a0c1204f8b02 Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Mon, 5 Mar 2018 11:49:25 +0530
Subject: [PATCH 0313/1852] drm: xlnx: mixer: Fixing wrong stride alignment

commit 836bfd26c61b375a94d7763111b3432ebbed9b7d from
https://github.com/Xilinx/linux-xlnx.git

Stride alignment must be 8 X pixels_per_clock value.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_mixer.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_mixer.c b/drivers/gpu/drm/xlnx/xlnx_mixer.c
index f0a3d540663e..8911cf676717 100644
--- a/drivers/gpu/drm/xlnx/xlnx_mixer.c
+++ b/drivers/gpu/drm/xlnx/xlnx_mixer.c
@@ -99,6 +99,7 @@
 #define	XVMIX_SCALE_FACTOR_2X		1
 #define	XVMIX_SCALE_FACTOR_4X		2
 #define	XVMIX_SCALE_FACTOR_INVALID	3
+#define	XVMIX_BASE_ALIGN		8
 
 /*************************** STATIC DATA  ************************************/
 static const u32 color_table[] = {
@@ -129,8 +130,6 @@ static const u32 color_table[] = {
 #define to_xlnx_crtc(x)	container_of(x, struct xlnx_crtc, crtc)
 #define to_xlnx_plane(x)	container_of(x, struct xlnx_mix_plane, base)
 #define to_xlnx_mixer(x)	container_of(x, struct xlnx_mix, crtc)
-#define get_xlnx_mixer_mem_align(m)  \
-	sizeof((m)->mixer_hw.layer_data[0].layer_regs.buff_addr1)
 
 /**
  * enum xlnx_mix_layer_id - Describes the layer by index to be acted upon
@@ -779,7 +778,7 @@ static unsigned int xlnx_mix_crtc_get_align(struct xlnx_crtc *crtc)
 	struct xlnx_mix_plane *plane = to_xlnx_plane(crtc->crtc.primary);
 	struct xlnx_mix *m = plane->mixer;
 
-	return get_xlnx_mixer_mem_align(m);
+	return XVMIX_BASE_ALIGN * m->mixer_hw.ppc;
 }
 
 /**
-- 
2.31.1

